maps/castle_wall.go

```diff
@@ -0,0 +1,642 @@
+package maps
+
+import (
+	"github.com/BattlesnakeOfficial/rules"
+)
+
+func init() {
+	globalRegistry.RegisterMap("hz_castle_wall", CastleWallMediumHazardsMap{})
+	globalRegistry.RegisterMap("hz_castle_wall_lg", CastleWallLargeHazardsMap{})
+	globalRegistry.RegisterMap("hz_castle_wall_xl", CastleWallExtraLargeHazardsMap{})
+}
+
+func setupCastleWallBoard(startingPositions [][]rules.Point, hazards []rules.Point, initialBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	rand := settings.GetRand(initialBoardState.Turn)
+
+	// place snakes
+	var snakes []rules.Point
+	// always support up to 8x snakes on all supported board sizes
+	snakes = append(snakes, startingPositions[0]...)
+	if len(initialBoardState.Snakes) >= 5 {
+		snakes = append(snakes, startingPositions[1]...)
+	}
+
+	// only support 8 or less snakes on boards smaller than XLarge
+	if (len(initialBoardState.Snakes) > 8) && (initialBoardState.Width < rules.BoardSizeXLarge) {
```

Comment:
I think you could pass in the maximum allowed snakes as a parameter for the function and just do one check and then have each map provide a value that is associated with the map. I would try my best to keep setupCastleWallBoard relatively independent of board size, you can handle that logic in the different maps themselves.

Reply:
Then you can just have one check that returns ErrorTooManySnakes